create-neon/test/create-neon.ts

```diff
@@ -0,0 +1,178 @@
+  }
+
+  add(lines: string[]) {
+    if (isCompleteLine(lines[lines.length - 1])) {
+      lines.push("");
+    }
+    this.buffer[this.buffer.length - 1] += lines.shift();
+    this.buffer = this.buffer.concat(lines);
+  }
+
+  find(p: (s: string) => boolean): string[] | null {
+    let index = this.buffer.findIndex(p);
+    if (index === -1) {
+      return null;
+    }
+    let extracted = this.buffer.splice(0, index + 1);
+    if (this.buffer.length === 0) {
+      this.buffer.push("");
+    }
+    return extracted;
+  }
+}
+
+async function* dialog(script: Record<string, string>, stdin: Writable, stdout: Readable) {
+  let lines = new LinesBuffer();
+
+  let keys = Object.keys(script);
+  let i = 0;
+  for await (let chunk of readChunks(stdout)) {
+    lines.add(splitLines(chunk));
+    let found = lines.find(line => line.startsWith(keys[i]));
+    if (found) {
+      stdin.write(script[keys[i]] + "\n");
+      yield found;
+      i++;
+      if (i >= keys.length) {
+        break;
+      }
+    }
+  }
+}
+
+const NODE: string = process.execPath;
+const CREATE_NEON = path.join(__dirname, '..', 'dist', 'src', 'bin', 'create-neon.js');
+
+describe('Command-line argument validation', () => {
+  it('requires an argument', async () => {
+    try {
+      await execa(NODE, [CREATE_NEON]);
+      assert.fail("should fail when no argument is supplied");
+    } catch (expected) {
+      assert.isTrue(true);
+    }
+  });
+
+  it('rejects extra arguments', async () => {
+    try {
+      await(execa(NODE, [CREATE_NEON, 'name', 'ohnoanextraargument']));
+      assert.fail("should fail when too many arguments are supplied");
+    } catch (expected) {
+      assert.isTrue(true);
+    }
+  });
+
+  it('fails if the directory already exists', async () => {
+    try {
+      await execa(NODE, [CREATE_NEON, 'dist']);
+      assert.fail("should fail when directory exists");
+    } catch (expected) {
+      assert.isTrue(true);
+    }
+  });
+});
+
+const PROJECT = 'create-neon-test-project';
+
+async function start(): Promise<ChildProcess> {
+  await rmdir(PROJECT, { recursive: true });
+  return spawn(NODE, [CREATE_NEON, PROJECT]);
+}
+
+/*
+function timeout(ms: number): Promise<void> {
+  let resolve: () => void;
+  let result: Promise<void> = new Promise(res => { resolve = res; });
+  setTimeout(() => { resolve() }, ms);
+  return result;
+}
+*/
+
+function exit(child: ChildProcess): Promise<number | null> {
+  let resolve: (code: number | null) => void;
+  let result: Promise<number | null> = new Promise(res => { resolve = res; });
+  child.on('exit', code => {
+    resolve(code);
+  });
+  return result;
+}
+
+const DEFAULTS_SCRIPT = {
+  'package name:':   '',
+  'version:':        '',
+  'description:':    '',
+  'git repository:': '',
+  'keywords:':       '',
+  'author:':         '',
+  'license:':        '',
+  'Is this OK?':     ''
+};
+
+describe('Project creation', () => {
+  it('succeeds with all default answers', async () => {
```

Comment:
This test fails if `npm run build` has not run. Should we add `"pretest": "npm run build"` to the package.json scripts?